llpc/builder/llpcBuilderCommon.h

```diff
@@ -0,0 +1,56 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcBuilderCommon.h
+ * @brief LLPC header file: contains common interface types in the LGC interface
+ ***********************************************************************************************************************
+ */
+#pragma once
+
+#include <stdint.h>
+
+namespace Llpc
+{
+
+/// Enumerates LLPC shader types.
+enum ShaderType : uint32_t
```

Comment:
If you couldn't change the definitions in vkgcDef.h and would like to rename this enum, I prefer naming it ShadingStage, which is similar to ShaderStage and has identical meanings or mappings. Shading stage is a terminology used in SPIR-V spec :)

Reply:
On second thoughts, I decided it would be less disruptive to keep the name ShaderStage, and have the two definitions differentiated by namespace. So I also did the change to move the whole of LGC into a new "lgc" namespace. I was intending to do that in a later commit anyway.